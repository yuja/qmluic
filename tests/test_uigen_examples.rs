pub mod common;

#[test]
fn test_gadget_properties() {
    insta::assert_snapshot!(common::translate_file("examples/GadgetProperties.qml").unwrap());
}

#[test]
fn test_item_views() {
    insta::assert_snapshot!(common::translate_file("examples/ItemViews.qml").unwrap());
}

#[test]
fn test_layout_flow() {
    insta::assert_snapshot!(common::translate_file("examples/LayoutFlow.qml").unwrap());
}

#[test]
fn test_main_window() {
    insta::assert_snapshot!(common::translate_file("examples/MainWindow.qml").unwrap());
}

#[test]
fn test_settings_dialog() {
    insta::assert_snapshot!(common::translate_file("examples/SettingsDialog.qml").unwrap());
}

#[test]
fn test_static_item_model() {
    insta::assert_snapshot!(common::translate_file("examples/StaticItemModel.qml").unwrap());
}

#[test]
fn test_various_layouts() {
    insta::assert_snapshot!(common::translate_file("examples/VariousLayouts.qml").unwrap());
}

#[test]
fn test_customwidget_common_my_dialog_button_box() {
    insta::assert_snapshot!(common::translate_file(
        "examples/customwidget/common/MyDialogButtonBox.qml"
    )
    .unwrap());
}

#[test]
fn test_customwidget_main_dialog() {
    insta::assert_snapshot!(
        common::translate_file("examples/customwidget/MainDialog.qml").unwrap()
    );
}

#[test]
fn test_customwidget_settings_form() {
    insta::assert_snapshot!(
        common::translate_file("examples/customwidget/SettingsForm.qml").unwrap()
    );
}

#[test]
fn test_thg_hg_email_dialog() {
    insta::assert_snapshot!(common::translate_file("examples/thg/HgEmailDialog.qml").unwrap());
}
