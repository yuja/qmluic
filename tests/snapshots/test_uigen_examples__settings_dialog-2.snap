---
source: tests/test_uigen_examples.rs
expression: ui_support_h
---
#pragma once
#include "ui_settingsdialog.h"

namespace UiSupport {
class SettingsDialog
{
public:
    SettingsDialog(QDialog *root, Ui::SettingsDialog *ui): root_(root), ui_(ui) {}

    void setup()
    {
        this->setupLabel4Alignment();
        this->setupLabel4Font();
        this->setupLabel5Font();
        this->setupButtonBoxAccepted();
        this->setupButtonBoxRejected();
        this->updateLabel4Alignment();
        this->updateLabel4Font();
        this->updateLabel5Font();
    }

private:
    enum class BindingIndex : unsigned {
        Label4Alignment,
        Label4Font,
        Label5Font,
    };

    void setupLabel4Alignment()
    {
        QObject::connect(this->ui_->textAlignmentEdit, &QComboBox::currentIndexChanged, this->root_, [this]() { this->updateLabel4Alignment(); });
    }

    void updateLabel4Alignment()
    {
#ifndef QT_NO_DEBUG
        constexpr unsigned index = static_cast<unsigned>(BindingIndex::Label4Alignment);
        Q_ASSERT_X(!(this->bindingGuard_[index >> 5] & (1U << (index & 0x1f))), __func__, "binding loop detected");
        this->bindingGuard_[index >> 5] |= (1U << (index & 0x1f));
#endif
        this->ui_->label4->setAlignment(this->evalLabel4Alignment());
#ifndef QT_NO_DEBUG
        this->bindingGuard_[index >> 5] &= ~(1U << (index & 0x1f));
#endif
    }

    Qt::Alignment evalLabel4Alignment()
    {
        int a0;
        bool a1;
        bool a2;
        bool a3;
        bool a4;
    b0:
        a0 = this->ui_->textAlignmentEdit->currentIndex();
        goto b2;
    b1:
        Q_UNREACHABLE();
    b2:
        a1 = a0 == 0;
        if (a1)
            goto b3;
        else
            goto b5;
    b3:
        return Qt::AlignLeft;
    b4:
        goto b6;
    b5:
        a2 = a0 == 1;
        if (a2)
            goto b6;
        else
            goto b8;
    b6:
        return Qt::AlignRight;
    b7:
        goto b9;
    b8:
        a3 = a0 == 2;
        if (a3)
            goto b9;
        else
            goto b11;
    b9:
        return Qt::AlignHCenter;
    b10:
        goto b12;
    b11:
        a4 = a0 == 3;
        if (a4)
            goto b12;
        else
            goto b14;
    b12:
        return Qt::AlignJustify;
    b13:
        goto b14;
    b14:
        return Qt::AlignLeft;
    b15:
        Q_UNREACHABLE();
    b16:
        Q_UNREACHABLE();
    }

    void setupLabel4Font()
    {
        QObject::connect(this->ui_->textFontFamilyEdit, &QFontComboBox::currentFontChanged, this->root_, [this]() { this->updateLabel4Font(); });
        QObject::connect(this->ui_->textFontSizeEdit, &QSpinBox::valueChanged, this->root_, [this]() { this->updateLabel4Font(); });
    }

    void updateLabel4Font()
    {
#ifndef QT_NO_DEBUG
        constexpr unsigned index = static_cast<unsigned>(BindingIndex::Label4Font);
        Q_ASSERT_X(!(this->bindingGuard_[index >> 5] & (1U << (index & 0x1f))), __func__, "binding loop detected");
        this->bindingGuard_[index >> 5] |= (1U << (index & 0x1f));
#endif
        this->ui_->label4->setFont(this->evalLabel4Font());
#ifndef QT_NO_DEBUG
        this->bindingGuard_[index >> 5] &= ~(1U << (index & 0x1f));
#endif
    }

    QFont evalLabel4Font()
    {
        QFont a0;
        QFont a1;
        int a2;
    b0:
        a0 = this->ui_->textFontFamilyEdit->currentFont();
        a1 = a0;
        a2 = this->ui_->textFontSizeEdit->value();
        a1.setPointSize(a2);
        return a1;
    b1:
        Q_UNREACHABLE();
    }

    void setupLabel5Font()
    {
        QObject::connect(this->ui_->textFontFamilyEdit, &QFontComboBox::currentFontChanged, this->root_, [this]() { this->updateLabel5Font(); });
        QObject::connect(this->ui_->textFontSizeEdit, &QSpinBox::valueChanged, this->root_, [this]() { this->updateLabel5Font(); });
    }

    void updateLabel5Font()
    {
#ifndef QT_NO_DEBUG
        constexpr unsigned index = static_cast<unsigned>(BindingIndex::Label5Font);
        Q_ASSERT_X(!(this->bindingGuard_[index >> 5] & (1U << (index & 0x1f))), __func__, "binding loop detected");
        this->bindingGuard_[index >> 5] |= (1U << (index & 0x1f));
#endif
        this->ui_->label5->setFont(this->evalLabel5Font(this->ui_->label5->font()));
#ifndef QT_NO_DEBUG
        this->bindingGuard_[index >> 5] &= ~(1U << (index & 0x1f));
#endif
    }

    QFont evalLabel5Font(QFont a)
    {
        a.setFamily(this->evalLabel5FontFamily());
        a.setPointSize(this->evalLabel5FontPointSize());
        return a;
    }

    QString evalLabel5FontFamily()
    {
        QFont a0;
        QString a1;
    b0:
        a0 = this->ui_->textFontFamilyEdit->currentFont();
        a1 = a0.family();
        return a1;
    }

    int evalLabel5FontPointSize()
    {
        int a0;
    b0:
        a0 = this->ui_->textFontSizeEdit->value();
        return a0;
    }

    void setupButtonBoxAccepted()
    {
        QObject::connect(this->ui_->buttonBox, &QDialogButtonBox::accepted, this->root_, [this]() { this->onButtonBoxAccepted(); });
    }

    void onButtonBoxAccepted()
    {
    b0:
        this->root_->accept();
        return;
    }

    void setupButtonBoxRejected()
    {
        QObject::connect(this->ui_->buttonBox, &QDialogButtonBox::rejected, this->root_, [this]() { this->onButtonBoxRejected(); });
    }

    void onButtonBoxRejected()
    {
    b0:
        this->root_->reject();
        return;
    }

    QDialog *const root_;
    Ui::SettingsDialog *const ui_;
#ifndef QT_NO_DEBUG
    quint32 bindingGuard_[1] = {0};
#endif

};
} // namespace UiSupport

